components/motor/gpiostepper/gpiostepper.go

```diff
@@ -402,7 +402,25 @@ func (m *gpioStepper) GoTo(ctx context.Context, rpm, positionRevolutions float64
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
 func (m *gpioStepper) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	// calculate delay between steps for the thread in the goroutine that we started in component creation
+	m.stepperDelay = time.Duration(int64(float64(time.Minute) / (math.Abs(rpm) * float64(m.stepsPerRotation))))
```

Comment:
I'll def add a comment

Reply:
I think we can do
```suggestion
	m.stepperDelay = time.Duration(time.Minute / (int64(math.Abs(rpm) * m.stepsPerRotation))
```
We need to cast to a integer anyway since the stepper steps are discrete, and it simplifies this line a little.
Also this is prime for abstracting out into testable math to avoid future breakages, like the math functions at the core of the gpio motor driver.

Reply:
That approach would lead to roundoff errors. For example, if stepsPerRotation is 1 and rpm is 1.5, the stepper delay would be 1 minute with Rand's approach, but it ought to be 0.67 minutes instead. Casting to int should be the last thing done, to retain precision.

Reply:
I guess I'm confused because I don't think we can pass a non-nanosecond time magnitude into time.Duration, and the other two values are integers, aren't we only losing decimal precision on rpm?

Reply:
Ohh, if `rpm` and `stepsPerRotation` are both integers already, you're right that there's no need to do any casting anywhere (maybe from int32 to int64). I had assumed we were casting certain parts to floats because other parts were floats already.

Reply:
Just to be clear `time.Minute` [value set at 60e9 nanosecond] and `m.stepsPerRotation` are integers, `rpm` is a float64, we're losing decimal precision on rpm

But this math should be equivalent as well and might be clearer
```go
time.Duration(
          int64 (
          float64(time.Minute /  m.stepsPerRotation) / math.Abs(rpm)
          )
```

Reply:
We have our greatest calculation error here when the delay we calculate is on the order of nanoseconds. We protect against that with the minimum stepper delay, and stepper motors driven by gpio drivers do not really move themselves on a per nanosecond basis 😄

Reply:
Actually, never mind, I mistakenly thought time.Minute would output an integer, it outputs a Duration, so we need to cast it to a int or float anyway.

Reply:
Regardless @martha-johnston my main ask for this line is to take this particular line out and test the output anyway.

Reply:
okay. I'll write a software test and test in person once I get the gpiostepper working